src/main/java/com/barterbay/app/repository/GoodRepository.java

```diff
@@ -0,0 +1,14 @@
+package com.barterbay.app.repository;
+
+import com.barterbay.app.domain.Good;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Optional;
+
+public interface GoodRepository extends JpaRepository<Good, Long> {
+
+  @Override
+  @EntityGraph(attributePaths = {"user"}, type = EntityGraph.EntityGraphType.LOAD)
+  Optional<Good> findById(Long aLong);
```

Comment:
Bad idea override general method `findById` and added EntityGraph there
If you need findById with EntityGraph - create a custom name, use HQL and add EntityGraph 
The method from JpaRepository - no need to override them, because they are widely used, and if you add eager loading for some of the - other team members probable will not about it, and they want Good without User, but they will get Good WITH user 

Why do you need eagerly fetch User for your good?
You GoodCreatedDTO does not contain any user DTO - why do you fetch user?)


![image](https://user-images.githubusercontent.com/69751517/182012530-a1c0121b-5fd1-45d9-b109-5f11d502cde9.png)